blocks/layout-grid/src/grid/higher-order.js

```diff
@@ -122,10 +122,19 @@ export function withSetPreviewDeviceType() {
 	return withDispatch( ( dispatch ) => {
 		return {
 			setPreviewDeviceType( type ) {
-				const { __experimentalSetPreviewDeviceType } =
-					dispatch( 'core/edit-site' ) ||
-					dispatch( 'core/edit-post' );
-				__experimentalSetPreviewDeviceType( type );
+				const isSiteEditor = document.querySelector(
+					'#edit-site-editor'
+				);
+
+				if ( isSiteEditor ) {
+					dispatch(
```

Comment:
We're missing a short-circuit here (`return`) in case we are inside the post editor -- otherwise it's going to dispatch two actions 😸

Apart from that, the code looks good 👍 

I wonder if we could extract this `isSiteEditor` variable to a function or a constant elsewhere so we could have a single source of truth and more reliability. Or, I don't know if we have a function already that performs this kind of check...

Reply:
> We're missing a short-circuit here (return) in case we are inside the site editor -- otherwise, it's going to dispatch two actions 😸

Oh, good catch! Updated that. It shouldn't have had any adverse side effects as using that post editor dispatch from the site editor should have no ill effect, but OMG once you pointed it out it smelled so stinky. 😆 

> I wonder if we could extract this isSiteEditor variable to a function or a constant elsewhere so we could have a single source of truth and more reliability. Or, I don't know if we have a function already that performs this kind of check...

I don't think there are any other functions like that in block experiments or need for them (but i could be wrong I don't work in this repo much), but it's not a bad idea.  I tried searching for something in the gutenberg stores that might supply this info but didn't find anything.

We also do a similar check for tracking in calyso's wpcom-block-editor here (https://github.com/Automattic/wp-calypso/blob/85e2d342cd9aa1aaf8fcfb8295928f71811d3be1/apps/wpcom-block-editor/src/wpcom/features/utils.js#L10), but id leave packaging that to be exported/imported and shared in both repos as a welcome follow up.

Reply:
Definitely 👍